packages/xdl/src/Android.ts

```diff
@@ -711,55 +721,100 @@ async function openUrlAsync({
   let installedExpo = false;
   let clientApplicationId = 'host.exp.exponent';
 
+  const installExpoIfNeeded = async (device: Device) => {
```

Comment:
Maybe add some form of automated testing since this logic was pretty fragile.

Reply:
I'll do it in a separate PR.